lib/ConfigFileProcessor.js

```diff
@@ -75,6 +80,79 @@ class ConfigFileProcessor {
       done(null, results.pop());
     });
   }
+
+  /**
+   * Reads a JSON configuration file and returns a JavaScript object. Does all the necessary processing and resolves all referenced files.
+   * @param {string} filePath
+   */
+  static readJSONConfigFile(file, cb) {
+    ConfigFileProcessor._readJSONConfigFile(file, 'root', (err, config) => {
+      logger.debug('Final configuration: ' + JSON.stringify(config, null, 4));
```

Comment:
What if there is an error here?

Reply:
Makes sense, we wouldn't want this to print if there is an error, I will put it in an "if".